torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -33,10 +33,12 @@ auto AccumulateGrad::acc_inplace(std::shared_ptr<Variable>& grad,
 
 auto AccumulateGrad::apply(const variable_list& grads) -> variable_list {
   // XXX: this method is not thread-safe!
-  check_input_variables("AccumulateGrad", grads, 1);
-  auto var = variable.lock();
+  check_input_variables("AccumulateGrad", grads, 1, 0);
   auto new_grad = grads[0];
 
+  if (!new_grad) return {};
```

Comment:
This is actually used in this PR because for example, when `gI` requires grad [here](https://github.com/albanD/pytorch/blob/clean_conv_double_back/torch/csrc/autograd/functions/convolution.cpp#L480-L505), it is non zero only if `ggW` is provided. So it happens that we pass null grads to AccumulateGrad, I think this is more efficient than creating a new tensor full of 0s.